python-avd/pyavd/_cv/workflows/deploy_to_cv.py

```diff
@@ -151,59 +128,40 @@ async def deploy_to_cv(
             # Create workspace
             await create_workspace_on_cv(workspace=result.workspace, cv_client=cv_client)
 
-            # Form deduplicated list of targeted CVDevices
-            devices = list(
-                {
-                    id(device): device
-                    for device in (
-                        [tag.device for tag in device_tags if tag.device is not None]
-                        + [tag.device for tag in interface_tags if tag.device is not None]
-                        + [config.device for config in configs if config.device is not None]
-                    )
-                }.values()
-            )
             # Check structured config of the targeted devices for overlapping `serial_number`s or `system_mac_address`es.
-            verify_device_inputs(devices, result.warnings, strict_system_mac_address=strict_system_mac_address)
+            verify_device_inputs(avd_devices, result.warnings, strict_system_mac_address=strict_system_mac_address)
+
+            # Form list of WorkflowDevice instance from given list of input AvdDevice instance.
+            # We will be then passing list of WorkflowDevice instance to rest of the wrappers
+            # to help identify changes to be done on CV side.
+            workflow_devices = build_workflow_device_inputs(avd_devices)
```

Comment:
No need to have a separate function in a different module. The comprehension can go here:
```suggestion
            workflow_devices = [WorkflowDevice.from_avd_input(avd_device) for avd_device in avd_devices]
```

Reply:
Sounds good.